src/QAT/qat/purr/backends/echo.py

```diff
@@ -115,6 +116,14 @@ def generate_connectivity(con_type, qubit_count):
     return None
 
 
+def add_direction_couplings_to_hardware(model, connectivity):
```

Comment:
This isn't echo specific. I know a few things around here aren't. Might be worth thinking about moving them somewhere else. :thinking:

Reply:
so not all setup functions add connectivity by default. For now the need rose for echo, once other needs come across, we can move to a common ground